src/notification.c

```diff
@@ -328,10 +346,12 @@ void notification_init(struct notification *n)
                 n->timeout = settings.timeouts[n->urgency];
 
         /* Icon handling */
-        if (STR_EMPTY(n->icon))
-                g_clear_pointer(&n->icon, g_free);
-        if (!n->raw_icon && !n->icon)
-                n->icon = g_strdup(settings.icons[n->urgency]);
+        if (STR_EMPTY(n->iconname))
+                g_clear_pointer(&n->iconname, g_free);
+        if (!n->icon && n->iconname)
+                n->icon = icon_get_for_name(n->iconname, &n->icon_id);
```

Comment:
Missed one here, this call doesn't scale the icon.

Reply:
OMG. Yes.

Reply:
> EDIT: I think this is a hint to maybe put all the icon parsing in one place to be able to manage them.

Valid point. This already exists. [I just had to use the right function](https://github.com/dunst-project/dunst/compare/5748c41124a9523f26414720da3df601d52d8611..6f8b53c4e8e0edc45758316dc3ccf006f931f86b).